### PR TITLE
chore: Tweaked .gitignore

### PR DESCRIPTION
**Pull request by danwilliams**
# Tweaked `.gitignore` file

  - Removed `Cargo.lock` and `pnpm-lock.yaml` from the `.gitignore` file, as these should not be committed.

  - Made `.gitignore` alphabetical.

## Summary by CodeRabbit

- **Chores**
	- Updated `.gitignore` to include/exclude specific directories and files.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,13 +1,16 @@
-.idea/
-res/
+Cargo.lock
```
Comment (fbozic) [inline .gitignore]: Why are we not committing `Cargo.lock`? Am I missing something big?

File: .gitignore [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
 target/
+target
 .DS_Store
+.idea/
 .vscode/
 
 # typescript
-node_modules
 *.map
+pnpm-lock.yaml
```
Comment (fbozic) [inline .gitignore]: Why are we not committing `pnmp-lock.yaml`? Am I missing something big?

Comment (danwilliams) [reply .gitignore]: @fbozic see discussion above 🙂 I hope that helps?

File: .gitignore (hunk 1 above)
Comment (danwilliams) [reply .gitignore]: @fbozic see discussion above 🙂 I hope that helps?

File: .gitignore (hunk 2 above)
Comment (fbozic) [reply .gitignore]: yeah, sorry missed it cos I've opened it from email directly on the review page

File: .gitignore (hunk 1 above)
Comment (fbozic) [reply .gitignore]: yeah, sorry missed it cos I've opened it from email directly on the review page

File: .gitignore [hunk 3]
```diff
@@ -1,13 +1,14 @@
-.idea/
-res/
 data/
+res/
 target/
+target
```
Comment (miraclx) [inline .gitignore:4]: duplicate

**Pull request merged**